applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,299 @@
+}
+
+// This function checks whether the element is cut by the wake
+const bool Define2DWakeProcess::CheckIfWakeElement(BoundedVector<double, 3>& rNodalDistancesToWake)
+{
+    // Initialize counters
+    unsigned int number_of_nodes_with_positive_distance = 0;
+    unsigned int number_of_nodes_with_negative_distance = 0;
+
+    // Count how many element nodes are above and below the wake
+    for (unsigned int i = 0; i < rNodalDistancesToWake.size(); i++) {
+        if (rNodalDistancesToWake(i) < 0.0) {
+            number_of_nodes_with_negative_distance += 1;
+        }
+        else {
+            number_of_nodes_with_positive_distance += 1;
+        }
+    }
+
+    // Elements with nodes above and below the wake are wake elements
+    return number_of_nodes_with_negative_distance > 0 &&
+           number_of_nodes_with_positive_distance > 0;
+}
+
+// This function adds the trailing edge elements in the
+// trailing_edge_sub_model_part
+const void Define2DWakeProcess::AddTrailingEdgeElements()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    std::sort(mTrailingEdgeElementsOrderedIds.begin(),
+              mTrailingEdgeElementsOrderedIds.end());
+    root_model_part.GetSubModelPart("trailing_edge_sub_model_part").AddElements(mTrailingEdgeElementsOrderedIds);
+}
+
+// This function selects the kutta elements. Kutta elements are touching the
+// trailing edge from below
+const void Define2DWakeProcess::MarkKuttaElements()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    ModelPart& trailing_edge_sub_model_part =
+        root_model_part.GetSubModelPart("trailing_edge_sub_model_part");
+
+    BoundedVector<double, 3> distance_to_element_center = ZeroVector(3);
+    double distance_to_wake = 1.0;
+
+    for (int i = 0;
+         i < static_cast<int>(trailing_edge_sub_model_part.Elements().size()); i++) {
+        ModelPart::ElementIterator it_elem =
+            trailing_edge_sub_model_part.ElementsBegin() + i;
+
+        // Compute the distance from the element's center to the trailing edge
+        distance_to_element_center =
+            ComputeDistanceFromTrailingEdgeToPoint(it_elem->GetGeometry().Center());
+
+        // Compute the projection of the distance vector in the wake normal
+        // direction
+        distance_to_wake = inner_prod(distance_to_element_center, mWakeNormal);
+
+        if (distance_to_wake < 0.0) {
+            it_elem->SetValue(KUTTA, true);
+        }
+    }
+}
+
+// This function finds the trailing edge element that is further downstream and
+// marks it as wake trailing edge element. The rest of trailing edge elements
+// are unassigned from the wake.
+const void Define2DWakeProcess::MartkWakeTrailingEdgeElement()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    ModelPart& trailing_edge_sub_model_part =
+        root_model_part.GetSubModelPart("trailing_edge_sub_model_part");
+
+    for (int i = 0;
```

Comment:
I think this loop can be parallelized.

Reply:
I don't think it is worth it due to the small amount of elements in the trailing_edge_sub_model_part. Actually I measured a loss in performance when I parallelized it